### DISCUSSION TITLE
Integrate custom class documentation with built-in documentation

### DISCUSSION THREAD

**Discussion by SIsilicon** [godot-extended-libraries/godot-ideas#36] [Ideas]
## Context
Code documentation

## Problem
As a programmer, there are four steps in making a program.
1. Plan it
2. Code it
3. Debug it (probably go back to step 2.)
4. Document it

That last step is especially important when making a library that others use. Now in Godot, the built-in classes have documentation embbeded in, and can be easily searched for from within the editor.
Custom made classes are a different story. You either resort to making external documentation which leaves room for out of date info, or use plain old comments which require users to dig into your code. Both options aren't as good as what godot does in my opinion.

## Suggestion

We create a docs generator, that reads annotated comments in the code, and display them the same way the built-in classes are.  Not only will it be easier to maintain the documentation, but the user will have a more streamlined experience with consulting the docs. No need to switch tabs. No need to read code.

## Solution

Achieving this would require injecting the classes' documentation data, which was parsed from the scripts, into the search help popup. Then it would have to be written similar to how the built-in ones are.

**Comment by SIsilicon** [comment]
I'm currently working on the right now.

**Comment by Calinou** [comment]
This was discussed in a proposal which is now implemented in 4.0: https://github.com/godotengine/godot-proposals/issues/993

Given it relies on the GDScript rewrite, it can't be backported to `3.x` though.
